src/persistent/_timestamp.c

```diff
@@ -166,12 +163,38 @@ TimeStamp_init_gmoff(void)
         return -1;
     }
 
-    gmoff = TimeStamp_abst(t->tm_year + TS_BASE_YEAR, t->tm_mon, t->tm_mday - 1,
-               t->tm_hour * 60 + t->tm_min, t->tm_sec);
+    *p_gmoff = _abst(
+        t->tm_year + TS_BASE_YEAR,
+        t->tm_mon,
+        t->tm_mday - 1,
+        t->tm_hour * 60 + t->tm_min,
+        t->tm_sec
+    );
 
     return 0;
 }
 
+/*
+ *  Timestamp type
+ */
+
+typedef struct
+{
+    PyObject_HEAD
+    /*
+      The first four bytes of data store the year, month, day, hour, and
+      minute as the number of minutes since Jan 1 00:00.
+
+      The final four bytes store the seconds since 00:00 as
+      the number of microseconds.
+
+      Both are normalized into those four bytes the same way with
+      TS_[UN]PACK_UINT32_INTO|FROM_BYTES.
+    */
+
+    unsigned char data[8];
+} TimeStamp;
+
 static void
 TimeStamp_dealloc(TimeStamp *ts)
```

Comment:
For heap-based types, I think we need the GC handling as discussed in https://docs.python.org/3.11/howto/isolating-extensions.html#garbage-collection-protocol